lib/taurus/core/taurusmanager.py

```diff
@@ -120,15 +125,28 @@ def addJob(self, job, callback=None, *args, **kw):
         :param args: (list) list of arguments passed to the job
         :param kw: (dict) keyword arguments passed to the job
         """
-        if self._serialization_mode == TaurusSerializationMode.Concurrent:
+        if kw.has_key("taurus_serialization_mode"):
+            serialization_mode = kw.pop("taurus_serialization_mode")
+        else:
+            serialization_mode = self._serialization_mode
+
+        if serialization_mode == TaurusSerializationMode.Concurrent:
             if not hasattr(self, "_thread_pool") or self._thread_pool is None:
```

Comment:
the `hasattr` check is not needed anymore since you are now always initializing the pool.

Reply:
We saw that, but we repeated the same protection that _thread_pool has.